File: sdk/ruby/test/wire/channel_endpoints_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ChannelEndpointsWireTest < WireMockTestCase
  def setup
    super

    @client = AgentMailer::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_channel_endpoints_list_with_wiremock
    test_id = "channel_endpoints.list.0"

    @client.channel_endpoints.list(request_options: {
      additional_headers: {
        "X-Test-Id" => "channel_endpoints.list.0"
      }
    })

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/channel-endpoints",
      query_params: nil,
      expected: 1
    )
  end

  def test_channel_endpoints_create_with_wiremock
    test_id = "channel_endpoints.create.0"

    @client.channel_endpoints.create(
      inbox_id: "inboxId",
      pod_id: "podId",
      channel: "sms",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "channel_endpoints.create.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/channel-endpoints",
      query_params: nil,
      expected: 1
    )
  end

  def test_channel_endpoints_get_with_wiremock
    test_id = "channel_endpoints.get.0"

    @client.channel_endpoints.get(
      endpoint_id: "endpointId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "channel_endpoints.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/channel-endpoints/endpointId",
      query_params: nil,
      expected: 1
    )
  end

  def test_channel_endpoints_update_with_wiremock
    test_id = "channel_endpoints.update.0"

    @client.channel_endpoints.update(
      endpoint_id: "endpointId",
      status: "active",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "channel_endpoints.update.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "PATCH",
      url_path: "/v1/channel-endpoints/endpointId",
      query_params: nil,
      expected: 1
    )
  end
end

File: sdk/rust/tests/channel_endpoints_test.rs
use agentmailer::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_channel_endpoints_list_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client.channel_endpoints.list(None).await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/channel-endpoints", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_channel_endpoints_create_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .channel_endpoints
        .create(
            &ChannelEndpointCreate {
                inbox_id: "inboxId".to_string(),
                pod_id: "podId".to_string(),
                channel: ChannelEndpointCreateChannel::Sms,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/channel-endpoints", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_channel_endpoints_get_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .channel_endpoints
        .get(&"endpointId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/channel-endpoints/endpointId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_channel_endpoints_update_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .channel_endpoints
        .update(
            &"endpointId".to_string(),
            &ChannelEndpointUpdate {
                status: ChannelEndpointUpdateStatus::Active,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("PATCH", "/v1/channel-endpoints/endpointId", None, 1)
        .await
        .unwrap();
}
